readthedocs/core/utils/general.py

```diff
@@ -23,3 +31,17 @@ def wipe_version_via_slugs(version_slug, project_slug):
     ]
     for del_dir in del_dirs:
         broadcast(type='build', task=remove_dirs, args=[(del_dir,)])
+
+    _clear_html_files_from_cloud_storage(version)
+
+
+def _clear_html_files_from_cloud_storage(version):
+    """Removes html files from cloud storage for a given version of a project."""
```

Comment:
```suggestion
    """Removes html files from media storage (cloud or local) for a given version of a project."""
```

Reply:
I should also mention that there is a [very similar function/task](https://github.com/readthedocs/readthedocs.org/blob/9125deb6c39516529759a73e1e57f7702aefb6b9/readthedocs/projects/tasks.py#L1839) to this already if you want to make it async.

Reply:
Yes... async will be great.